### PR TITLE
Replace GPL'd colorize gem with MIT-licensed rainbow

### PR DESCRIPTION
**Pull request by jamesc**
colorize is licensed under GPL-2 which conflicts with the MIT license of
github_changelog_generator.  This changes all usage of colorize to
rainbow which does have a compatible license (MIT)

### CODE REVIEW

**Comment by olleolleolle** [comment]
Hello James! 

Good choice of library.

I wonder: we're coloring strings that pass through our `log` function in the `Helper` namespace. Idea: Can we ditch all spread-out coloring of these?

**Comment by olleolleolle** [comment]
@jamesc Hi! I made my comment into a PR-onto-your-PR.

**Comment by olleolleolle** [comment]
@jamesc Oh, I made a change, again.

File: github_changelog_generator.gemspec [hunk 1]
```diff
@@ -27,6 +27,9 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "github_api", ">= 0.12"
   spec.add_runtime_dependency "rainbow", ">= 2.1"
 
+  # rack 2.0 does not work on Ruby < 2.2.  Lock it down here
+  spec.add_runtime_dependency "rack", "~> 1.2"
```
Comment (olleolleolle) [inline github_changelog_generator.gemspec]: This is contentious: if people are using this library in their application, we're holding the rack back.

**Comment by jamesc** [comment]
Yeah, I backed out that change.  It doesn't fix the travis builds anyway.

**Comment by olleolleolle** [comment]
@jamesc Cool, thanks for your hard effort!

This change fixes a hard problem, and we got the logging mechanism placed more centrally, by only using its formatter. And we got rid of the String extension, which the colorize gem also had.

To make the PR really stand out in the changelog, do you want to rephrase the title of it? Since we will be making a CHANGELOG.md out of this, something clearer (for understanding at a glance). Examples include:

- Use rainbow instead of colorize
- Replace GPL'd colorize gem with MIT-licensed rainbow

**Comment by olleolleolle** [comment]
Again, thanks a lot for doing this, @jamesc! 🎉

**Pull request merged**